### PR TITLE
Move kafkareceiver to contrib repo

### PR DESCRIPTION
**Pull request by punya**
Fixes #3639.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -13,6 +13,10 @@ updates:
     directory: "/exporter/alibabacloudlogserviceexporter"
     schedule:
       interval: "weekly"
+  - package-ecosystem: "gomod"
+    directory: "/exporter/awscloudwatchlogsexporter"
+    schedule:
+      interval: "weekly"
```
Comment (bogdandrutu) [inline .github/dependabot.yml:19]: Consider to run the script in a different PR?

File: cmd/otelcontribcol/components.go [hunk 2]
```diff
@@ -123,6 +124,7 @@ func components() (component.Factories, error) {
 		fluentforwardreceiver.NewFactory(),
 		influxdbreceiver.NewFactory(),
 		jmxreceiver.NewFactory(),
+		kafkareceiver.NewFactory(),
```
Comment (bogdandrutu) [inline cmd/otelcontribcol/components.go:127]: Consider to plug this in a separate PR

File: internal/testdata/common.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testdata
+
+import (
+	"go.opentelemetry.io/collector/model/pdata"
+)
```
Comment (bogdandrutu) [inline internal/testdata/common.go:19]: May be a good time to start thinking of exposing this as "pdatatest"?

**Comment by punya** [comment]
Abandoning this in favor of working on healthcheckextension instead.

**Pull request closed without merging**